hare/hare.go

```diff
@@ -368,6 +370,13 @@ func (h *Hare) onTick(ctx context.Context, lid types.LayerID) (bool, error) {
 		nonce = &nnc
 	}
 
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	if len(h.cps) >= h.config.LimitConcurrent {
+		err := fmt.Errorf("failed to start cp for layer %q hare reached concurrent cp limit %q", lid.Uint32(), h.config.LimitConcurrent)
+		logger.With().Warning("hare failure", log.Err(err))
+		return false, err
```

Comment:
this is incorrect. what we want is to kill the oldest consensus process and make room for the new one. later consensus is more important than old ones. if the old one fails locally, it's likely we can already get a block certificate from the network if other nodes succeed. 

i'd suggest leaving that as a TODO earlier. i think it's better doing that in a separate PR.

Reply:
Yep this is wrong, but trying to re-instate the old logic, which was also broken would be quite a lot of work and I'm not seeing that it's worth doing that just to have a separate PR, when the fix here would be quite simple.

Reply:
sure if you feel comfortable doing so.

i just want to caution that changing too much behavior in one PR is not rollback friendly. but up to you.